Common/ColorConv.h

```diff
@@ -0,0 +1,167 @@
+}
+
+inline u16 BGRA8888toRGB565(u32 px) {
+	return ((px >> 19) & 0x001F) | ((px >> 5) & 0x07E0) | ((px << 8) & 0xF800);
+}
+
+inline u16 BGRA8888toRGBA4444(u32 px) {
+	return ((px >> 20) & 0x000F) | ((px >> 8) & 0x00F0) | ((px << 4) & 0x0F00) | ((px >> 16) & 0xF000);
+}
+
+inline u16 BGRA8888toRGBA5551(u32 px) {
+	return ((px >> 19) & 0x001F) | ((px >> 6) & 0x03E0) | ((px << 7) & 0x7C00) | ((px >> 16) & 0x8000);
+}
+
+inline u16 RGBA8888toRGBA5551(u32 px) {
+	return ((px >> 3) & 0x001F) | ((px >> 6) & 0x03E0) | ((px >> 9) & 0x7C00) | ((px >> 16) & 0x8000);
+}
+
+inline u32 RGBA2BGRA(u32 src) {
+	const u32 r = (src & 0x000000FF) << 16;
+	const u32 ga = src & 0xFF00FF00;
+	const u32 b = (src & 0x00FF0000) >> 16;
+	return r | ga | b;
+}
+
+inline u32 DecodeRGBA4444(u16 src) {
+	const u32 r = (src & 0x000F) << 0;
+	const u32 g = (src & 0x00F0) << 4;
+	const u32 b = (src & 0x0F00) << 8;
+	const u32 a = (src & 0xF000) << 12;
+
+	const u32 c = r | g | b | a;
+	return c | (c << 4);
+}
+
+inline u32 DecodeRGBA5551(u16 src) {
+	u8 r = Convert5To8((src >> 0) & 0x1F);
+	u8 g = Convert5To8((src >> 5) & 0x1F);
+	u8 b = Convert5To8((src >> 10) & 0x1F);
+	u8 a = (src >> 15) & 0x1;
+	a = (a) ? 0xff : 0;
+	return (a << 24) | (b << 16) | (g << 8) | r;
+}
+
+inline u32 DecodeRGB565(u16 src) {
+	u8 r = Convert5To8((src >> 0) & 0x1F);
+	u8 g = Convert6To8((src >> 5) & 0x3F);
+	u8 b = Convert5To8((src >> 11) & 0x1F);
+	u8 a = 0xFF;
+	return (a << 24) | (b << 16) | (g << 8) | r;
+}
+
+inline u32 DecodeRGBA8888(u32 src) {
+#if 1
+	return src;
+#else
+	// This is the order of the bits.
+	u8 r = src & 0xFF;
+	u8 g = (src >> 8) & 0xFF;
+	u8 b = (src >> 16) & 0xFF;
+	u8 a = (src >> 24) & 0xFF;
+	return (a << 24) | (b << 16) | (g << 8) | r;
+#endif
+}
+
+inline u16 RGBA8888To565(u32 value) {
+	u8 r = value & 0xFF;
+	u8 g = (value >> 8) & 0xFF;
+	u8 b = (value >> 16) & 0xFF;
+	r >>= 3;
+	g >>= 2;
+	b >>= 3;
+	return (u16)r | ((u16)g << 5) | ((u16)b << 11);
+}
+
+inline u16 RGBA8888To5551(u32 value) {
+	u8 r = value & 0xFF;
+	u8 g = (value >> 8) & 0xFF;
+	u8 b = (value >> 16) & 0xFF;
+	u8 a = (value >> 24) & 0xFF;
+	r >>= 3;
+	g >>= 3;
+	b >>= 3;
+	a >>= 7;
+	return (u16)r | ((u16)g << 5) | ((u16)b << 10) | ((u16)a << 15);
+}
+
+inline u16 RGBA8888To4444(u32 value) {
+	const u32 c = value >> 4;
+	const u16 r = (c >> 0) & 0x000F;
+	const u16 g = (c >> 4) & 0x00F0;
+	const u16 b = (c >> 8) & 0x0F00;
+	const u16 a = (c >> 12) & 0xF000;
+	return r | g | b | a;
+}
+
+// convert image to 8888, parallelizable
+// TODO: Implement these in terms of the conversion functions below.
+void convert4444_gl(u16* data, u32* out, int width, int l, int u);
+void convert565_gl(u16* data, u32* out, int width, int l, int u);
+void convert5551_gl(u16* data, u32* out, int width, int l, int u);
+void convert4444_dx9(u16* data, u32* out, int width, int l, int u);
+void convert565_dx9(u16* data, u32* out, int width, int l, int u);
+void convert5551_dx9(u16* data, u32* out, int width, int l, int u);
```

Comment:
I wonder if we should keep these separate, or if it makes sense to have all the funcs use this format?  Hmm.

-[Unknown]